src/Shell32.Desktop/Shell32+ITEMIDLIST.cs

```diff
@@ -0,0 +1,27 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System.Runtime.InteropServices;
+
+    /// <content>
+    /// Contains the <see cref="ITEMIDLIST"/> nested type.
+    /// </content>
+    public partial class Shell32
+    {
+        /// <summary>
+        /// Contains a list of item identifiers.
+        /// </summary>
+        /// <remarks>Used by <see cref="SHGetKnownFolderIDList"/> and <see cref="SHGetFolderLocation"/></remarks>
+        [StructLayout(LayoutKind.Sequential)]
+        public struct ITEMIDLIST
+        {
+            /// <summary>
+            /// A list of item identifiers.
+            /// </summary>
+            [MarshalAs(UnmanagedType.Struct)]
```

Comment:
You mentioned you couldn't use pointers in your P/Invoke method signatures for this struct. This is why. You can't use pointers to structs that have `MarshalAs` attributes in them.